tensorboard/plugins/graph/tf_graph_board/tf-graph-board.ts

```diff
@@ -309,4 +310,29 @@ class TfGraphBoard extends LegacyElementMixin(PolymerElement) {
       ? node.include
       : tf_graph.InclusionType.UNSPECIFIED;
   }
+  @observe('graph')
+  _slimGraphChanged() {
+    // By default, avoid coloring by 'structure' for large graphs, since it may be very expensive.
+    // Color by 'structure' is still available to users via explicit gesture.
+    if (!this.graph) {
+      return;
+    }
+    const isGraphTooLarge =
+      Object.keys(this.graph.nodes).length > 10000 &&
```

Comment:
Would you mind parameterizing this via module const or graph params? You can do it on a follow up.